Tests/GRPCCoreTests/Configuration/MethodConfigurationCodingTests.swift

```diff
@@ -0,0 +1,424 @@
+      XCTAssertEqual(json, expected)
+    }
+  }
+
+  func testDecodeInvalidProtobufDuration() throws {
+    for timestamp in ["1", "1ss", "1S", "1.0S"] {
+      let json = "\"\(timestamp)\""
+      try self.testDecodeThrowsRuntimeError(json: json, as: GoogleProtobufDuration.self)
+    }
+  }
+
+  func testDecodeRPCCodeFromCaseName() throws {
+    let inputs: [(String, Status.Code)] = [
+      ("OK", .ok),
+      ("CANCELLED", .cancelled),
+      ("UNKNOWN", .unknown),
+      ("INVALID_ARGUMENT", .invalidArgument),
+      ("DEADLINE_EXCEEDED", .deadlineExceeded),
+      ("NOT_FOUND", .notFound),
+      ("ALREADY_EXISTS", .alreadyExists),
+      ("PERMISSION_DENIED", .permissionDenied),
+      ("RESOURCE_EXHAUSTED", .resourceExhausted),
+      ("FAILED_PRECONDITION", .failedPrecondition),
+      ("ABORTED", .aborted),
+      ("OUT_OF_RANGE", .outOfRange),
+      ("UNIMPLEMENTED", .unimplemented),
+      ("INTERNAL", .internalError),
+      ("UNAVAILABLE", .unavailable),
+      ("DATA_LOSS", .dataLoss),
+      ("UNAUTHENTICATED", .unauthenticated),
+    ]
+
+    for (name, expected) in inputs {
+      let json = "\"\(name)\""
+      let code = try self.decoder.decode(GoogleRPCCode.self, from: Data(json.utf8))
+      XCTAssertEqual(code.code, expected)
+    }
+  }
+
+  func testDecodeRPCCodeFromRawValue() throws {
+    let inputs: [(Int, Status.Code)] = [
+      (0, .ok),
+      (1, .cancelled),
+      (2, .unknown),
+      (3, .invalidArgument),
+      (4, .deadlineExceeded),
+      (5, .notFound),
+      (6, .alreadyExists),
+      (7, .permissionDenied),
+      (8, .resourceExhausted),
+      (9, .failedPrecondition),
+      (10, .aborted),
+      (11, .outOfRange),
+      (12, .unimplemented),
+      (13, .internalError),
+      (14, .unavailable),
+      (15, .dataLoss),
+      (16, .unauthenticated),
+    ]
+
+    for (rawValue, expected) in inputs {
+      let json = "\(rawValue)"
+      let code = try self.decoder.decode(GoogleRPCCode.self, from: Data(json.utf8))
+      XCTAssertEqual(code.code, expected)
+    }
+  }
+
+  func testEncodeDecodeRPCCode() throws {
+    let codes: [Status.Code] = [
+      .ok,
+      .cancelled,
+      .unknown,
+      .invalidArgument,
+      .deadlineExceeded,
+      .notFound,
+      .alreadyExists,
+      .permissionDenied,
+      .resourceExhausted,
+      .failedPrecondition,
+      .aborted,
+      .outOfRange,
+      .unimplemented,
+      .internalError,
+      .unavailable,
+      .dataLoss,
+      .unauthenticated,
+    ]
+
+    for code in codes {
+      let encoded = try self.encoder.encode(GoogleRPCCode(code: code))
+      let decoded = try self.decoder.decode(GoogleRPCCode.self, from: encoded)
+      XCTAssertEqual(decoded.code, code)
+    }
+  }
+
+  func testDecodeRetryPolicy() throws {
+    let json = """
+      {
+        "maxAttempts": 1,
+        "initialBackoff": "1s",
+        "maxBackoff": "3s",
+        "backoffMultiplier": 1.6,
+        "retryableStatusCodes": ["ABORTED"]
```

Comment:
Bit of a nit, but maybe we could have at least one test with multiple codes here to make sure decoding works well in that scenario too?